### DISCUSSION TITLE
how to use Connect-PnPOnline in a server without internet connection

### DISCUSSION THREAD

**Discussion by spoorthienganti** [pnp/powershell#812] [Q&A]
Hi All,

I am trying to use the PNP PowerShell in a server where there is no internet connection. I saved the module on the machine which has internet access and then moved the files to the server, below are the commands I used 

Save-Module -Name PnP.PowerShell -RequiredVersion 1.6.0 -Path [C:\PNPPowershellModuleInstall]

I moved the files and used the Import module on the server machine

Import-Module C:\PNPPowershellModuleInstall\PnP.PowerShell\1.6.0\PnP.PowerShell.psd1 -DisableNameChecking

then I tired to connect to a sharepoint online site using the below command

Connect-PnPOnline  {onlinesiteurl} -UseWebLogin -ReturnConnection

which tried to load the authentication page of the site and as the server don't have internet connection failed to load

Could you please let me know how to proceed

Thanks.

**Comment by LeonArmston** [comment]
Is this to authenticate on the server to SharePoint On-Premise? If so the new PnP.PowerShell only runs for SharePoint online and requires an internet connection.

You will need to download the old PnP PowerShell Module SharePointPnPPowerShell201X, which is still available but is no longer maintained.

Instructions & releases are here 
[https://github.com/pnp/PnP-PowerShell/releases](https://github.com/pnp/PnP-PowerShell/releases)
